dimod/reference/samplers/exact_solver.py

```diff
@@ -20,14 +20,18 @@
     energy for every possible sample, they are very slow.
 """
 import numpy as np
+from itertools import product
 
 from dimod.core.sampler import Sampler
-from dimod.sampleset import SampleSet, as_samples
+from dimod.sampleset import SampleSet
 from dimod.core.polysampler import PolySampler
 from dimod.vartypes import Vartype
-from dimod.exceptions import SamplerUnknownArgWarning
 
-__all__ = ['ExactSolver', 'ExactPolySolver', 'ExactDQMSolver']
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
```

Comment:
btw, you might not actually need this `if` branch, I was assuming you would but worth checking explicitly

Reply:
Double checked, the `if` is necessary - thanks for the tip!